tests/limits/test_vif_limit.py

```diff
@@ -0,0 +1,104 @@
+import pytest
+
+import ipaddress
+import logging
+import os
+import tempfile
+
+from pkgfixtures import host_with_saved_yum_state
+
+# Requirements:
+# - one XCP-ng host (--host) >= 8.2
+# - a VM (--vm)
+# - the first network on the host can be used to reach the host
+
+vif_limit = 16
+interface_name = "enX"
+vcpus = '8'
+
+# There is a ResourceWarning due to background=True on an ssh call
+# We do ensure the processes are killed
+@pytest.mark.filterwarnings("ignore::ResourceWarning")
+@pytest.mark.debian_uefi_vm
+class TestVIFLimit:
+    def test_vif_limit(self, host_with_saved_yum_state, imported_vm):
+        host = host_with_saved_yum_state
+        vm = imported_vm
+        if (vm.is_running()):
+            logging.info("VM already running, shutting it down first")
+            vm.shutdown(verify=True)
+
+        network_uuid = vm.vifs()[0].param_get('network-uuid')
+        existing_vifs = len(vm.vifs())
+
+        logging.info(f'Get {vcpus} vCPUs for the VM')
+        original_vcpus_max = vm.param_get('VCPUs-max')
+        original_vcpus_at_startup = vm.param_get('VCPUs-at-startup')
+        vm.param_set('VCPUs-max', vcpus)
+        vm.param_set('VCPUs-at-startup', vcpus)
+
+        logging.info('Create VIFs before starting the VM')
+        vifs = []
+        for i in range(existing_vifs, vif_limit):
+            vif = vm.create_vif(i, network_uuid=network_uuid)
+            vifs.append(vif)
+
+        vm.start()
+        vm.wait_for_os_booted()
+
+        logging.info('Verify the interfaces exist in the guest')
+        for i in range(0, vif_limit):
+            if vm.ssh_with_result([f'test -d /sys/class/net/{interface_name}{i}']).returncode != 0:
+                guest_error = vm.ssh_with_result(['dmesg | grep -B1 -A3 xen_netfront']).stdout
+                logging.error("dmesg:\n%s", guest_error)
+                assert False, "The interface does not exist in the guest, check dmesg output above for errors"
+
+        logging.info('Configure interfaces')
+        config = '\n'.join([f'iface {interface_name}{i} inet dhcp\n'
+                            f'auto {interface_name}{i}'
+                            for i in range(existing_vifs, vif_limit)])
+        vm.ssh([f'echo "{config}" >> /etc/network/interfaces'])
+
+        logging.info('Install iperf3 on VM and host')
+        if vm.ssh_with_result(['apt install iperf3 --assume-yes']).returncode != 0:
+            assert False, "Failed to install iperf3 on the VM"
+        host.yum_install(['iperf3'])
+
+        logging.info('Reconfigure VM networking')
+        if vm.ssh_with_result(['systemctl restart networking']).returncode != 0:
+            assert False, "Failed to configure networking"
+
+        # Test iperf on all interfaces in parallel
```

Comment:
I feel like benchmarking should be in a separate test suite from the routine tests, as benchmarks are not meant to verify an update train (as is the purpose of complete runs).

Reply:
Yes, we agree.